transcom/mymove
# [MB-10466] REFINE Same Service Items are being requested for payment-Prime bug

## [Jira ticket](https://dp3.atlassian.net/browse/MB-10466?atlOrigin=eyJpIjoiNTMwNDM0OGFiNjEyNDgyNmI0NjhlZWViMmQ2ZjczM2UiLCJwIjoiaiJ9)

## Summary
This is a refinement ticket for the previous [PR](https://github.com/transcom/mymove/pull/11079) that was made for this. 
This PR further covers caveats that prime users may come across while creating a payment request. 
Function to check status of existing payment requests will be in `rules.go`. The new function accounts for Move Level items (i.e. Move management & Counseling), which don't require a shipment/shipmentID. 
Also had to update `services/move_task_order/move_task_order_fetcher` to eager load MTOServiceItems ReService Code.
Multiple tests had to be refined and updated due to the new function. Tests will be in 
1) `rules_test.go`
2) `payment_creator_test.go`
3) `prime/api/move_task_order_test.go`
4) `prime/api/payment_request_test.go`
5) `prime/apiv2/move_task_order_test.go`
6) `payment_request_recalculator_test.go`


## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

1) Go to officelocal, click "Local Sign In", and log in as Prime simulator.
2) Select a move and click on "Create Payment Request" (blue button at the top, in the Moves section).
3) Scroll down to "Shipment Service Items", go to "Domestic origin 1st day SIT", click the checkbox "Add to payment request".
4) Scroll down and click "Submit Payment Request".
5) You should get routed back to the previous details page and see a green banner at the top saying it was successful.
6) Click on "Create Payment Request" again for the same move. 
7) Select the same service item you selected previously.
8) This time, you should see a red error banner saying "Conflict error".

### Backend


## Screenshots
<img width="1211" alt="Screenshot 2023-08-09 at 8 10 09 AM" src="https://github.com/transcom/mymove/assets/110994428/888ecb36-426f-4865-874d-8bf37f0b0b88">
